### ISSUE TITLE
Now playing indicator on multiple songs

### ISSUE THREAD

**Issue by C4-x** [fast4x/RiMusic#4956]
### Version of RiMusic

0.6.64

### Android version

12

### I've done these:

- [X] Uninstall old build before installing a new one
- [X] Checked opening issues about my similar problem

### Tell us how you trigger this bug

1. Open RiMusic
2. Play a song from a playlist or album or search results 
3. Click on the next song in the playlist/album/search results 
4. Multiple now playing indicators appear 

### Submit a video

If you are on Android, please download a screen recorder and record your screen while you
duplicating the problem as describe above.

> You may need to use Github App to be able to upload video - size limit is 10 MB.

If you are on Desktop, drag video into this box and it will be uploaded automatically
![Screenshot_20241218-163401](https://github.com/user-attachments/assets/7064478f-a1cc-4bbf-ad39-8f5721edd30e)
![Screenshot_20241214-154352](https://github.com/user-attachments/assets/95a52b3e-ce23-4299-8207-cea833bd2f3c)

### What was your expectation?

Now playing indicator should be only one the song currently Playing

**Comment by twistios** [comment]
https://github.com/user-attachments/assets/afc06a99-4a3b-4da8-8f5d-4aaa882acebe

The problem does not happen for me in the custom version.
https://github.com/fast4x/RiMusic/tree/master/customVersion

**Comment by C4-x** [comment]
> You can also see that none of the indicators actually are at the correct song.

Yes, most of the times it's incorrect like in the screenshot below, How do I fix it if it is device specific ? 

Logs attached below 

![Screenshot_20241218-221935.jpg](https://github.com/user-attachments/assets/76fd22cc-3103-47ea-8896-17100f6ae2ec)



2024-12-18 22:20:58:949 DEBUG: MainApplication - Log enabled at /data/user/0/it.fast4x.rimusic/files/logs
2024-12-18 22:20:59:743 DEBUG: T - PlayerServiceModern QueuePersistentEnabled Saved queue
2024-12-18 22:21:01:305 DEBUG: h - Success loadData in QuickPicsModern
2024-12-18 22:21:05:476 DEBUG: T - PlayerServiceModern QueuePersistentEnabled Saved queue
2024-12-18 22:21:07:532 DEBUG: PlayerServiceModern - PlayerServiceModern onPositionDiscontinuity oldPosition 5 newPosition 8 reason 4
2024-12-18 22:21:07:562 DEBUG: T - PlayerServiceModern QueuePersistentEnabled Saved queue
2024-12-18 22:21:11:326 DEBUG: PlayerServiceModern - PlayerServiceModern onPositionDiscontinuity oldPosition 8 newPosition 6 reason 4
2024-12-18 22:21:11:358 DEBUG: T - PlayerServiceModern QueuePersistentEnabled Saved queue

**Comment by twistios** [comment]
It probably is a bug, I just wanted to show that it does not simply reproduce just like you showed.

**Comment by coolcarp** [comment]
I have a similar issue. If I skip a song too quickly, then the indicator will lag behind on the previous songs. However, if I wait 5 seconds into the song, then skip, the indicator will show up correctly in the Queue. Rotating to landscape also corrects it.
